qualtran/bloqs/block_encoding/product_test.py

```diff
@@ -63,11 +68,41 @@ def test_product_signature():
     )
 
 
+@frozen
+class TestPrepareOracle(PrepareOracle):
+    @property
+    def selection_registers(self) -> Tuple[Register, ...]:
+        return (Register('z', QBit()),)
+
+    @property
+    def junk_registers(self) -> Tuple[Register, ...]:
+        return (Register('a', QAny(5)),)
+
+
+@frozen
+class TestBlockEncoding(BlockEncoding):
+    alpha: float = 1
+    epsilon: float = 0
+    system_bitsize: int = 1
+    ancilla_bitsize: int = 1
+    resource_bitsize: int = 1
+
+    @property
+    def signature(self) -> Signature:
+        return Signature.build(system=1, ancilla=1, resource=1)
+
+    @property
+    def signal_state(self) -> BlackBoxPrepare:
+        return BlackBoxPrepare(TestPrepareOracle())
+
+
```

Comment:
move them to `bloqs/for_testing/`?

Reply:
I could try in a follow-up PR. The thing is that these really are stub classes whose only purpose is to test edge cases of constructors. They're not really useful for independent testing of any other behavior.